weave/trace/weave_client.py

```diff
@@ -2018,6 +2019,14 @@ def flush(
             callback: Optional callback function that receives status updates.
                       Overrides use_progress_bar.
         """
+        if self._server_is_flushable:
+            server = cast(RemoteHTTPTraceServer, self.server)
+            if not server.call_processor.is_accepting_new_work():
+                raise ValueError(
+                    "client.finish() called while async processor already stopping. client.finish() "
```

Comment:
`async processor` is an implementation detail. I would instead logger.warn("`finish` has already been called") and return early

Reply:
i removed this, as we will now always start up another thread after the client forces a flush

Reply:
so its okay if we call finish multiple times.

Reply:
but also you are right, bad error message